src/packs/checker/dependency.rs

```diff
@@ -25,6 +25,39 @@ impl ValidatorInterface for Checker {
             node_to_pack.insert(node, pack);
         }
 
+        let mut add_edge = |from_pack: &Pack, to_pack: &Pack| {
+            let from_node = pack_to_node
+                .get(&from_pack)
+                .expect("Could not find from_pack")
+                .to_owned();
+            let to_node = pack_to_node
+                .get(&to_pack)
+                .expect("Could not find to_pack")
+                .to_owned();
+            graph.add_edge(from_node, to_node, ());
+        };
+        let mut error_messages: Vec<String> = vec![];
+        let mut validate_architecture =
```

Comment:
I'm thinking this validator check should live in `architecture.rs` instead of `dependency.rs`, what do you think? We might need to extract some common helpers to keep things DRY.